neo3/wallet/account.py

```diff
@@ -0,0 +1,258 @@
+        """
+        Instantiate and returns an account from a given private key and password.
+
+        Args:
+            private_key: the private key that will be used to create an encrypted key.
+            password: the password to encrypt a randomly generated private key.
+
+        Returns:
+            The newly created account.
+        """
+        return cls(password=password, private_key=private_key)
+
+    @classmethod
+    def watch_only(cls, script_hash: types.UInt160) -> Account:
+        """
+        Instantiate and returns a watch-only account from a given script hash.
+
+        Args:
+            script_hash: the script hash that will identify an account to be watched.
+
+        Returns:
+            The account that will be monitored.
+        """
+        return cls(password='', watch_only=True, address=cls.script_hash_to_address(script_hash))
+
+    @classmethod
+    def watch_only_from_address(cls, address: str) -> Account:
+        """
+        Instantiate and returns a watch-only account from a given address.
+
+        Args:
+            address: the address that will identify an account to be watched.
+
+        Returns:
+            The account that will be monitored.
+        """
+        return cls(password='', watch_only=True, address=address)
+
+    @staticmethod
+    def script_hash_to_address(script_hash: types.UInt160) -> str:
+        """
+        Converts the specified script hash to an address.
+
+        Args:
+            script_hash: script hash to convert
+        """
+        version = settings.network.account_version  # this is the current Neo's protocol version
+        data_ = version.to_bytes(1, 'little') + script_hash.to_array()
+
+        return base58.b58encode_check(data_).decode('utf-8')
+
+    @staticmethod
+    def address_to_script_hash(address: str) -> types.UInt160:
+        """
+        Converts the specified address to a script hash.
+
+        Args:
+            address: address to convert
+
+        Raises:
+            ValueError: if the length of data_ isn't 21.
+            ValueError: if the script hash version isn't 3.0.
+        """
+        data_ = base58.b58decode_check(address)
+        if len(data_) != len(types.UInt160.zero()) + 1:
+            raise ValueError('The address is wrong, because data_ length should be 21')
+
+        if data_[0] != settings.network.account_version:   # Only accepted version is 3.0
+            raise ValueError('The version is not 3.0')
+
+        return types.UInt160(data=data_[1:])
+
+    @staticmethod
+    def private_key_from_nep2(nep2_key: str, passphrase: str) -> bytes:
+        """
+        Decrypt a nep2 key into a private key.
+
+        Args:
+            nep2_key: the key that will be decrypt.
+            passphrase: the password to decrypt the nep2 key.
+
+        Raises:
+            ValueError: if the passphrase is incorrect or the version of the account is not 3.0.
```

Comment:
1. The docstring regarding the version is a bit confusing. The account version has to be `0x35` (or 53d), not specifically `3.0`. The `3.0` is the required version for the `Wallet`.
2. The code also raises a ValueError if key decoding fails